### PR TITLE
Added changes to clean up the container after a SIGINT

### PR DESCRIPTION
**Pull request by bharathappali**
This PR fixes the issue #82 

Using `trap` to run `ctrlc_handler` function on a `SIGINT` and clean up the docker containers created.

changes `2` to `&` to avoid `stdout` getting printed in `docker stop` command

Signed-off-by: bharathappali <bharath.appali@gmail.com>

### CODE REVIEW

**Comment by bharathappali** [comment]
As i couldn't add anyone as reviewers i'm mentioning request for reviewers below 

@dinogun @ShishirH

**Comment by bharathappali** [comment]
Missed to see `-t` option in deploy script. `Ctrl+C` along with `deploy.sh -t -c docker` solves the issue.

Closing this PR as its not required

**Comment by ShishirH** [comment]
While right now only docker terminate works, I believe we should be cleaning up for every environment in the future. In such a case, I think `trap "{cluster_type}_terminate" 2` would be better

**Comment by bharathappali** [comment]
@ShishirH Yeah we can add that in the `ctrlc_handler` as in future we may get any other pre-cleanup stuff needs to be done. I'm fine in doing either ways, jus thought having a seperate function will be a good approach.

File: scripts/docker-helpers.sh [hunk 1]
```diff
@@ -176,7 +176,7 @@ function docker_start() {
 
 function docker_terminate() {
 	echo -n "###   Uninstalling kruize for docker..."
-	docker stop kruize grafana prometheus cadvisor 2>/dev/null
+	docker stop kruize grafana prometheus cadvisor &> /dev/null
```
Comment (dinogun) [inline scripts/docker-helpers.sh]: Why is this change needed ? We might run into issues such as [this](https://unix.stackexchange.com/questions/356408/strange-problem-with-trap-and-sigint) if we background the process.

File: deploy.sh [hunk 2]
```diff
@@ -50,6 +50,20 @@ user="admin"
...
+    if [[ "$cluster_type" == "docker" ]]; then
+        # Terminate the containers [kruize && grafana && prometheus && cadvisor]
+        docker_terminate
+    fi
+    # Exiting gracefully
+    exit 2
+}
+
+
+trap "ctrlc_handler" 2 # Handle SIGINT for cleaning up containers in docker case
```
Comment (dinogun) [inline deploy.sh]: We can add the same handler for other signals as well (SIGHUP, SIGINT, SIGQUIT)

File: scripts/docker-helpers.sh (hunk 1 above)
Comment (bharathappali) [reply scripts/docker-helpers.sh:179]: @dinogun  Thanks for the review. 

Aah! Sorry i haven't thought about, When i was testing my PR i saw `stdout` of docker stop (which is the name of containers stopped) I felt it a bit unaligned as the `	echo -n "###   Uninstalling kruize for docker..."` doesn't produce an new line and the output looked as 

```
###   Uninstalling kruize for docker...kruize
grafana
prometheus
cadvisor
done
```

So jus backgrounded the process and directed `stdout` as well to `/dev/null`

Will redirecting `stderr` to `stdout` and `stdout` to `/dev/null` be a good solution ?

File: deploy.sh (hunk 2 above)
Comment (bharathappali) [reply deploy.sh:64]: Sure. Will be adding it.

File: scripts/docker-helpers.sh (hunk 1 above)
Comment (bharathappali) [reply scripts/docker-helpers.sh]: reverted back to previous changes

File: deploy.sh (hunk 2 above)
Comment (bharathappali) [reply deploy.sh]: @dinogun made the changes as we discussed. Can i please have it reviewed ?
